### PR TITLE
Handle multi-byte unicode characters in json parsing

### PR DESCRIPTION
**Pull request by Tymolc**
Multi-byte characters (e.g. Emojis) in JSONs could not be parsed. Parsing failed with error `UTF-16 string is missing low surrogate`.
The code now checks if the first byte is a "High Surrogate" byte, which suggests, that there is a following byte.

Related to #139, #469, #635 (to some extend),

### CODE REVIEW

**Comment by Tymolc** [comment]
Doing some research on this, I still think it is the more common way to assume UTF-16 BE here.  https://tools.ietf.org/html/rfc2781 (4.3) states that big-endian should be assumed if no BOM is available. Furthermore, JSON is usually UTF-8 encoded (apart from the `\u` escaped characters), so there is no way to implicitly assume an endianness.
As a reference, the JSON parser of [nlohmann](https://github.com/nlohmann/json) also [assumes big endianness](https://github.com/nlohmann/json/blob/e89c946451cfdb7392b12bc6c3674b548ea5c0ee/include/nlohmann/detail/input/lexer.hpp#L295).

File: Release/src/json/json_parsing.cpp [hunk 1]
```diff
@@ -682,26 +722,31 @@ inline bool JSON_Parser<CharType>::handle_unescape_char(Token& token)
...
-                const int isxdigitResult = _isxdigit_l(ch_int, utility::details::scoped_c_thread_locale::c_locale());
-#else
-                const int isxdigitResult = isxdigit(ch_int);
-#endif
-                if (!isxdigitResult) return false;
+                return false;
+            }
+
+            // handle multi-block characters that start with a high-surrogate
+            if (decoded > H_SURROGATE_START && decoded < H_SURROGATE_END)
```
Comment (reneme) [inline Release/src/json/json_parsing.cpp]: While comparing to nlohmann's jsonformoderncpp, we realized that `H_SURROGATE_START` and `H_SURROGATE_END` should be checked for as well, like so:

```suggestion
            if (decoded >= H_SURROGATE_START && decoded <= H_SURROGATE_END)
```

**Comment by Naville** [comment]
@BillyONeal can confirm this PR now solves the aforementioned issues. Is there any plan to merge this patch?
Thanks

**Comment by BillyONeal** [comment]
I agree, I read this wrong before. Thanks for your contribution!

**Pull request merged**